notebooks/Distance.ipynb

```diff
@@ -0,0 +1,812 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Initialize algorithm\n",
+    "GUndirected = G.toUndirected()\n",
+    "ctd = nk.distance.CommuteTimeDistance(GUndirected)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run\n",
+    "ctd.run()\n",
+    "# Alternatively \n",
+    "# ctd.runSinglePair(source,target)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# The distance from source to target node\n",
+    "print(ctd.distance(source, target))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Diameter "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This algorithm gives an estimation of the diameter of a given graph. The algorithm is based on the ExactSumSweep algorithm presented in Michele Borassi, Pierluigi Crescenzi, Michel Habib, Walter A. Kosters, Andrea Marino, Frank W. Takes[1](http://www.sciencedirect.com/science/article/pii/S0304397515001644).\n",
+    "\n",
+    "The [Diameter(G, algo=DiameterAlgo.Automatic, error=1.0, nSamples=0)](https://networkit.github.io/dev-docs/python_api/distance.html?highlight=diameter#networkit.distance.Diameter) constructor expects a graph as mandatory parameter. `algo` specifies the choice of diameter algorithm while `error` is the maximum allowed relative error. Set to 0 for the exact diameter. `nSamples`is the number of samples to be used. `algo` can be chosen between from \n",
+    "    0. automatic\n",
+    "    1. exact\n",
+    "    2. estimatedRange\n",
+    "    3. estimatedSamples\n",
+    "    4. estimatedPedantic\n",
+    "\n",
+    "Note that the input graph must be connected, otherwise the resulting diameter will be infinite. As the graph we are using is not connected, we shall extract the largest connected component from it and then compute the diameter of the resulting graph."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Extract largest connect component\n",
+    "newGraph = nk.components.ConnectedComponents.extractLargestConnectedComponent(GUndirected, True)\n",
+    "newGraph.numberOfNodes()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Initialize algorithm\n",
+    "diam = nk.distance.Diameter(newGraph,algo=1)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Run\n",
+    "diam.run()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Get diameter of graph\n",
```

Comment:
Mention why the returned diameter is a pair (lower bound, upper bound), and that in the exact case the diameter is the first value of the pair.